website_sale_vat_required/views/templates.xml

```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="address" inherit_id="website_sale.address" name="Address Management">
+        <xpath expr="//input[@name='vat']/.." position="replace"/>
+        <xpath expr="//input[@name='company_name']/../.." position="after">
+            <t t-if="mode[1] == 'billing'">
+                <div t-attf-class="form-group #{error.get('vat') and 'has-error' or ''} col-md-6 div_vat">
```

Comment:
Why do you need this?

Reply:
Good question. It seems [included upstream already](https://github.com/odoo/odoo/blob/d2b7be055ad3ca2223d4e8009388c80e5057dbf2/addons/website_sale/views/templates.xml#L1162-L1166).

Reply:
This migration has been made from v9 directly . In v9 the only thing you need to do is set 'vat' as mandatory field using _get_mandatory_billing_fields method. In v10 and v11, if you only do that and you don't modify the template 'address', then, when an user is logged in, odoo will not permit to complete the form until this field is filled, but this field will be hide and the user will wonder what happened. This is, because now, odoo only show this field if there is no user logged in.

Reply:
That's interesting. So, Odoo doesn't let a user modify his VAT number. IMHO that makes sense; if you need to change such field, you're probably better creating a new customer.

Then maybe you should only require `vat` when he's adding a new address. I think you can check that by making sure there's no `request.params["partner_id"]`.